File: src/main/kotlin/jimin/66week/혼자서 하는 틱택토.py
def solution(board):
    x = 0
    o = 0
    for i in range(3):
        for j in range(3):
            if board[i][j] == 'X':
                x += 1
            elif board[i][j] == 'O':
                o += 1

    if x > o:
        return 0
    if x == 0 and o == 0:
        return 1
    if o - x > 1:
        return 0

    x_bingo = 0
    o_bingo = 0
    for i in range(3):
        # 가로
        first = board[i][0]
        if first == '.':
            continue

        isBingo = True
        for j in range(1, 3):
            if first != board[i][j]:
                isBingo = False
                break

        if isBingo:
            if first == 'X':
                x_bingo += 1
            elif first == 'O':
                o_bingo += 1

    for i in range(3):
        # 세로
        first = board[0][i]
        if first == '.':
            continue

        isBingo = True
        for j in range(1, 3):
            if first != board[j][i]:
                isBingo = False
                break

        if isBingo:
            if first == 'X':
                x_bingo += 1
            elif first == 'O':
                o_bingo += 1

    # 대각선
    if board[0][0] != '.' and board[0][0] == board[1][1] and board[1][1] == board[2][2]:
        if board[0][0] == 'X':
            x_bingo += 1
        elif board[0][0] == 'O':
            o_bingo += 1
    elif board[0][2] != '.' and board[0][2] == board[1][1] and board[1][1] == board[2][0]:
        if board[0][2] == 'X':
            x_bingo += 1
        elif board[0][2] == 'O':
            o_bingo += 1

    if o_bingo > x_bingo and o == x + 1:
        return 1
    if o_bingo < x_bingo and o == x:
        return 1
    if o_bingo == 0 and x_bingo == 0:
        return 1

    return 0

File: src/main/kotlin/byeonghee/week66/혼자서 하는 틱택토.kt
package byeonghee.week66

class 소병희_혼자서_하는_틱택토 {
    fun solution(board: Array<String>): Int {
        var oCnt = 0
        var xCnt = 0

        for(i in 0 until 3) for(j in 0 until 3) {
            when(board[i][j]) {
                'O' -> oCnt++
                'X' -> xCnt++
            }
        }

        var oBingo = countBingo(board, 'O')
        var xBingo = countBingo(board, 'X')

        if (oBingo > 1 && xBingo > 1) return 0
        if (oBingo >= 1 && oCnt - xCnt != 1) return 0
        if (xBingo >= 1 && oCnt - xCnt != 0) return 0
        if (oCnt > xCnt + 1) return 0
        if (xCnt > oCnt) return 0

        return 1
    }

    fun countBingo(board: Array<String>, player: Char): Int {
        var answer = 0

        for(i in 0 until 3) {
            if (board[i][0] == player && board[i][0] == board[i][1] && board[i][1] == board[i][2]) answer++
            if (board[0][i] == player && board[0][i] == board[1][i] && board[1][i] == board[2][i]) answer++
        }

        if (board[1][1] == player && board[0][0] == board[1][1] && board[1][1] == board[2][2]) answer++
        if (board[1][1] == player && board[2][0] == board[1][1] && board[1][1] == board[0][2]) answer++

        return answer
    }
}

fun main() {
    val sol = 소병희_혼자서_하는_틱택토()
    val ret = sol.solution(
            arrayOf("OOO", "O..", "XXX")
    )
    println(ret)
}